zcommon/src/org/zkoss/lang/Library.java

```diff
@@ -41,7 +41,11 @@
  */
 public class Library {
 	private static final Logger log = LoggerFactory.getLogger(Library.class);
-	private static final Map<String, List<String>> _props = new HashMap<String, List<String>>();
+	//Using the ConcurrentHashMap to substitute synchronized HashMap can avoid the thread lock issue.
+	//Enhance the performance of the code. The reference is as follow. 
+	//1. https://dzone.com/articles/java-7-hashmap-vs
+	//2. https://stackoverflow.com/questions/1291836/concurrenthashmap-vs-synchronized-hashmap
+	private static final ConcurrentMap<String, List<String>> _props = new ConcurrentHashMap<String, List<String>>();
```

Comment:
1. You don't have to add comments for references - they will find ZK-3823 and you wrote there.
2. ` Map<String, List<String>>` is sufficient - you don't use any methods that ConcurrentMap provides